Packages/Tracking/Hands/Runtime/Scripts/HandModelManager.cs

```diff
@@ -0,0 +1,512 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+namespace Leap.Unity.HandsModule
+{
+    /// <summary>
+    /// A script to manage the active hand models in the scene
+    /// </summary>
+    public class HandModelManager : MonoBehaviour
+    {
+        [System.Serializable]
+        public class HandModelPair
+        {
+            public string HandModelPairId;
+            public HandModelBase Left;
+            public HandEnableDisable LeftEnableDisable;
+            public HandModelBase Right;
+            public HandEnableDisable RightEnableDisable;
+
+            public bool ContainsHandModel(HandModelBase handModel)
+            {
+                return Left == handModel || Right == handModel;
+            }
+        }
+
+        public Action<HandModelBase> OnHandModelEnabled;
+        public Action<HandModelBase> OnHandModelDisabled;
+
+        public List<HandModelPair> handModelPairs;
+
+        private void Reset()
+        {
+            handModelPairs = new List<HandModelPair>();
+            RegisterAllUnregisteredHandModels();
+        }
+
+        /// <summary>
+        /// Find any unregistered hand models, and register them with the hand model manager
+        /// </summary>
+        public void RegisterAllUnregisteredHandModels()
+        {
+            HandModelBase[] unpairedHandModels = FindObjectsOfType<HandModelBase>(true);
+
+            for (int i = 0; i < unpairedHandModels.Length; i++)
+            {
+                if (!IsRegistered(unpairedHandModels[i]))
+                {
+                    RegisterHandModel(unpairedHandModels[i]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Check if a hand model is registered with the Hand Model Manager
+        /// </summary>
+        /// <param name="handModel">The hand model to check</param>
+        /// <returns>Returns true if the hand model is registered, and false if not</returns>
+        public bool IsRegistered(HandModelBase handModel)
```

Comment:
How many of these methods can actually be private as they are only of use internally to the class?

Reply:
I think all public methods aside from this one are sensible to be exposed? The aim was to give a large amount of flexibility to developers to enable and disable hand models, and to register & unregister hand models if needed.

Reply:
OK, let's make this one internal and leave the others as they are?

Reply:
👍